dtm0/it/s3setup/s3setup.sh

```diff
@@ -0,0 +1,327 @@
+    # Uncomment and change MOTR_DEVEL_WORKDIR_PATH in motr sysconfig
+    # to use development files.
+    sed -i "s|^#MOTR_DEVEL_WORKDIR_PATH=.*|MOTR_DEVEL_WORKDIR_PATH=${MOTR_DIR}|" $MOTR_SYSCONFIG
+    patch $CLUSTER_YAML < $CLUSTER_YAML_PATCH
+}
+
+function motr_build()
+{
+    pushd $MOTR_DIR
+    ./scripts/install-build-deps
+    MAKE_OPTS=$MOTR_MAKE_OPTS CONFIGURE_OPTS=$MOTR_CONFIGURE_OPTS ./scripts/m0 rebuild
+    popd
+}
+
+function hare_build()
+{
+    pushd $HARE_DIR
+    patch -p1 < $HARE_PATCH
+    make
+    popd
+}
+
+function motr_install()
+{
+    pushd $MOTR_DIR
+    ./scripts/install-motr-service -l
+    rm -f /usr/lib64/libmotr-helpers.so.0 /usr/lib64/libmotr.so.2
+    ln -s "${MOTR_DIR}/helpers/.libs/libmotr-helpers.so.0" /usr/lib64/libmotr-helpers.so.0
+    ln -s "${MOTR_DIR}/motr/.libs/libmotr.so.2" /usr/lib64/libmotr.so.2
+    ln -s "${MOTR_DIR}/extra-libs/galois/src/.libs/libgalois-0.1.so" /usr/lib64/libgalois-0.1.so
+    popd
+}
+
+function hare_install()
+{
+    pushd $HARE_DIR
+    make devinstall    
+    popd
+}
+
+function cluster_bootstrap()
+{
+    hctl bootstrap --mkfs $CLUSTER_YAML
+}
+
+function cluster_stop()
+{
+    hctl shutdown
+}
+
+function s3bench_run()
+{
+    local ak
+    local sk
+    local sgiam_pass
+
+    # Generate password.
+    sgiam_pass=$(s3cipher decrypt --data="$(s3confstore properties:///opt/seagate/cortx/auth/resources/authserver.properties getkey --key ldapLoginPW)" --key="$(s3cipher generate_key --const_key cortx)")
+
+    # Fetch the user from ldap, copy access key (ak) and secret key (sk) from output.
+    read ak sk <<< $(ldapsearch -b "ou=accesskeys,dc=s3,dc=seagate,dc=com" -x -w $sgiam_pass -D "cn=sgiamadmin,dc=seagate,dc=com" "(&(objectclass=accesskey))" | egrep "^ak:|^sk:" | awk '{ print $2; }')
+
+    # Get s3bench.
+    if [[ ! -e ./s3bench ]]; then
+        wget $S3BENCH_URL -O ./s3bench
+        chmod +x ./s3bench
+    fi
+
+    # Run s3bench.
+    ./s3bench -accessKey $ak -accessSecret "${sk}" -bucket test1 -endpoint http://127.0.0.1  -numClients 4 -numSamples 4 -objectSize 1Mb
+}
+
+function fids_hex_get()
+{
+    local svcs_json_out=$(hctl status --json | jq -r '.nodes[] | .svcs[]')
+    local svc_json_out=$(echo $svcs_json_out | jq -r 'select( .name | contains("ioservice"))')
+    local s3_json_out=$(echo $svcs_json_out | jq -r 'select( .name | contains("s3server"))')
+    M0D_FIDS_HEX=($(echo $svc_json_out | jq -r '.fid' | sed -E 's/0x720+([0-9][:]0x[A-Za-z0-9]+)/\1/'))
+    S3_FID_HEX=$(echo $s3_json_out | jq -r '.fid' | sed -E 's/0x720+([0-9][:]0x[A-Za-z0-9]+)/\1/')
+}
+
+function pids_get()
+{
+    local pids=""
+    local pid
+
+    for fid in ${M0D_FIDS_HEX[@]} ; do
+        pid=$(ps ax | grep m0d | grep $fid | awk '{ print $1; }')
+        M0D_PIDS+=($pid)
+        pids+="$pid "
+    done
+
+    S3_PID=$(ps ax | grep s3server | grep $S3_FID_HEX | awk '{ print $1; }')
+
+    echo "m0d PIDs: $pids"
+    echo "s3server PID: $S3_PID"
+}
+
+function addb_dump()
```

Comment:
this foo() looks so similar to all2all's. can it be generalized?

Reply:
Done.